### ISSUE TITLE
[feature req] Please provide a full responsive (and toch optimized) backend with 1.10 - its about time ;)

### ISSUE THREAD

**Issue by lhwparis** [osTicket/osTicket-1.8#2170]
[feature req] Please provide a full responsive (and toch optimized) backend with 1.10 - its about time ;)

**Comment by ntozier** [comment]
The 1.10 DPR is already out, so I highly doubt that a feature that is slated for a future release (my understanding is that is one of the benchmarks for the 2.0 release) will be rushed to be added to the current version this late in the development cycle.  I do not feel that your request includes a realistic time frame.

**Comment by timomayer** [comment]
hi @ntozier thanks for your reply. i agree its too late in this development cycle but this opens the question to me why this was not considered much earlier for this cycle cause i am absolute sure mobile/touch are an as big usecase for osticket as it is for the whole web. 

anyway it is as it is. but i encourage you to consider a "fully mobile first responsive template for front and backend" for the next possible relase.

if you need help for the frontend part i can spend some time to help me. pls ping me.

**Comment by greezybacon** [comment]
I don't disagree that the project needs a responsive design. I mean, after all, *everybody* is doing responsive design these days, so why are we? right?

But I think this highlights a much bigger issue. osTicket needs to support varying "themes" so that different users and uses of the system can apply differing layouts, including responsive ones. 

One of the main hangups for the responsive design is because we the main developers can't seem to agree with the community on what a refined user interface sporting responsiveness should look like. I think this is another hint that themes are needed.

Currently, we are working on extending the core functionality rather than improving the user interface. I think UI theming will come early next year in the 2.0 release

**Comment by lhwparis** [comment]
Thanks for your Explanation! I agree a temple engine and flexible Templates are the Best was so everybody Can customize it to his own needs. If you ask the Community about the importance of this i am very sure it will be higher rated as many other core Features. I am sure an up2date User Experience is the key Feature for many Community members and not so much more Features.

**Comment by greezybacon** [comment]
@lhwparis unfortunately a man can't make a living working for free, and the feedback we get from our subscribers is primarily that there is missing functionality from the system. We'll get to the look and feel

**Comment by ghost** [comment]
Almost 5 years on and still not responsive.

**Comment by JediKev** [comment]
@groovybluedog This is slated for v2.0. If you are unhappy with the time, you can develop it yourself and make a pull.

**Comment by ghost** [comment]
@JediKev I wouldn't have a problem financially supporting the project to push things forward as I do not program, but even so, literally everything else is responsive now and this is the last thing left. It should have been a priority a lot sooner.

**Comment by ghost** [comment]
Still sad to see something so basic still not implemented in 2022. This is the only script I know of, that I would use, that is still not up to responsive standards. I too do not program and would financially contribute, but this is just silly.

**Comment by JediKev** [comment]
@Lewis8379 

Well you’d be ecstatic to know that this is currently in the works with v2.0 (full codebase rewrite). You can follow the v2.0 roadmap below:
- https://github.com/osTicket/roadmap

I can say that the new design is fully responsive (on all devices) and looks very beautiful and modern. We can’t wait to share screenshots and updates on the progress. Please stay tuned!

Cheers.

**Comment by ghost** [comment]
@JediKev I thought I had heard something similar. I am so glad it's coming, because osTicket is otherwise so AMAZING and feature rich tool. The only thing that holds it back so much is the lack of responsiveness. 

Will this responsiveness be on the front end, agent backend and admin backend?

The only other thing now is when v2 comes out. I see the block posts from last year, and I do understand how much of a change this will be, but I fear if it takes too long, then osTicket will fall behind in other features.

**Comment by zerosine** [comment]
Are there any modern/current plans for a Responsive design to come directly from the Creators?  Seems third-party solutions are the only method currently, which definitely breaks when an update is released, as OSTicket does not natively support 'themes'.

**Comment by zerosine** [comment]
large chunk of users are mobile/smartphone, if not the majority.

**Comment by JediKev** [comment]
@zerosine 

Yes, that’s one of the main focus points for the full codebase rewrite (v2.0) coming in the near future. v2.0 will be fully modern and responsive; even on mobile devices. There is no set release date so you’ll just have to follow this repo and the blog on the website for any updates on the release.

Cheers.

**Comment by ghost** [comment]
It should have been the focus point a long time ago. Sadly the lack of responsive theming is the major failing of osTicket that I have no doubt is holding a lot of people back. It could not come soon enough.
